pyproject.toml

```diff
@@ -2,7 +2,7 @@
 requires = ["setuptools",
             "setuptools_scm>=6.2",
             "cython==0.29.34",
-            "oldest-supported-numpy",
+            "numpy>=1.25,<2",
```

Comment:
I would recommend adding a temporary version cap on the runtime dependency as well, see https://github.com/scipy/oldest-supported-numpy/issues/76#issuecomment-1628819676

Reply:
Yes I mean `install_requires`, and it won't be blocking for future-compatibility tests because `2.0.0dev0` compares as `< 2.0` :)

Reply:
I guess I do not understand the logic of allowing 2.0.dev but preventing users from using the actual 2.0 later.

Reply:
Thanks for your concern, which is valid. But I would like to discuss it separate from this one. Can you please propose a PR at your convenience?

I don't mind max pinning numpy in release branches, but I would rather not do it in the main branch. I *think* scipy only pin maxversion when they branch for release (at least the last I look and I don't look very often).

Reply:
done at #15206 !